posthog/models/link.py

```diff
@@ -0,0 +1,108 @@
+import uuid
+
+from django.db import models
+from posthog.models.team import Team
+from posthog.models.utils import CreatedMetaFields, UpdatedMetaFields
+import structlog
+
+logger = structlog.get_logger(__name__)
+
+
+def generate_uuid():
+    return str(uuid.uuid4())
+
+
+class Link(CreatedMetaFields, UpdatedMetaFields, models.Model):
+    """
+    Links that redirect to a specified destination URL.
+    These are used for sharing URLs across the application.
+    """
+
+    id = models.CharField(max_length=36, primary_key=True, default=generate_uuid, editable=False)
+    redirect_url = models.URLField(max_length=2048)
+    short_link_domain = models.CharField(max_length=255, help_text="Domain where the short link is hosted, e.g. hog.gg")
+    short_code = models.CharField(
+        max_length=255, help_text="The unique code/path that identifies the short link, e.g. 'abc123'"
+    )
+    team = models.ForeignKey(
+        Team,
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        help_text="Optional association with a team, only for signed up customers",
```

Comment:
Make this required. We dont store non-signed-up links in Postgres, they go to a transient Redis instance instead.